src/sharepoint/rest/folders.ts

```diff
@@ -84,10 +149,44 @@ export class Folder extends QueryableInstance {
     }
 
     /**
-     * Gets this folder's files
+     * Gets a value that specifies the content type order.
      * 
      */
-    public get files(): Files {
-        return new Files(this);
+    public get uniqueContentTypeOrder(): QueryableCollection {
+        return new QueryableCollection(this, "UniqueContentTypeOrder");
     }
+
+    /**
+     * Gets this folder's welcome page
+     */
+    public get welcomePage(): Queryable {
+        return new Queryable(this, "WelcomePage");
+    }
+
+     /**
+     * Delete this folder
+     * 
+     * @param eTag Value used in the IF-Match header, by default "*"
+     */
+    public delete(eTag = "*"): Promise<void> {
+        return this.post({
+            headers: {
+                "IF-Match": eTag,
+                "X-HTTP-Method": "DELETE",
+            },
+        });
+    }
+
+    /**
+     * Moves the folder to the Recycle Bin and returns the identifier of the new Recycle Bin item.
+     */
+    public recycle(): Promise<string> {
+        this.append("recycle");
```

Comment:
Want to get away from the this.append, I would create a new File(this, "recycle") and return the .post() of that. I know you got this from my code, but I updated it to maintain the "this" instance - so you can blame me :)